### ISSUE TITLE
Fix debugging analysis UI from eclipse IDE

### ISSUE THREAD

**Issue by lfcnassif** [sepinf-inc/IPED#487]
Affects only developers. This stopped working months ago.

**Comment by lfcnassif** [comment]
I fixed some things, but still throwing this lucene error:
```
java.lang.NoSuchMethodError: org.apache.lucene.codecs.blocktree.BlockTreeTermsWriter.validateAutoPrefixSettings(II)V
	at org.apache.lucene.codecs.autoprefix.AutoPrefixPostingsFormat.<init>(AutoPrefixPostingsFormat.java:77)
	at org.apache.lucene.codecs.autoprefix.AutoPrefixPostingsFormat.<init>(AutoPrefixPostingsFormat.java:52)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at java.lang.Class.newInstance(Class.java:442)
	at org.apache.lucene.util.NamedSPILoader.reload(NamedSPILoader.java:72)
	at org.apache.lucene.util.NamedSPILoader.<init>(NamedSPILoader.java:51)
	at org.apache.lucene.util.NamedSPILoader.<init>(NamedSPILoader.java:38)
	at org.apache.lucene.codecs.PostingsFormat$Holder.<clinit>(PostingsFormat.java:49)
	at org.apache.lucene.codecs.PostingsFormat.forName(PostingsFormat.java:112)
	at org.apache.lucene.codecs.lucene62.Lucene62Codec.<init>(Lucene62Codec.java:167)
	at org.apache.lucene.codecs.lucene62.Lucene62Codec.<init>(Lucene62Codec.java:82)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at java.lang.Class.newInstance(Class.java:442)
	at org.apache.lucene.util.NamedSPILoader.reload(NamedSPILoader.java:72)
	at org.apache.lucene.util.NamedSPILoader.<init>(NamedSPILoader.java:51)
	at org.apache.lucene.util.NamedSPILoader.<init>(NamedSPILoader.java:38)
	at org.apache.lucene.codecs.Codec$Holder.<clinit>(Codec.java:47)
	at org.apache.lucene.codecs.Codec.forName(Codec.java:116)
	at org.apache.lucene.index.SegmentInfos.readCodec(SegmentInfos.java:424)
	at org.apache.lucene.index.SegmentInfos.readCommit(SegmentInfos.java:356)
	at org.apache.lucene.index.SegmentInfos.readCommit(SegmentInfos.java:288)
	at org.apache.lucene.index.StandardDirectoryReader$1.doBody(StandardDirectoryReader.java:57)
	at org.apache.lucene.index.StandardDirectoryReader$1.doBody(StandardDirectoryReader.java:54)
	at org.apache.lucene.index.SegmentInfos$FindSegmentsFile.run(SegmentInfos.java:692)
	at org.apache.lucene.index.StandardDirectoryReader.open(StandardDirectoryReader.java:77)
	at org.apache.lucene.index.DirectoryReader.open(DirectoryReader.java:63)
	at dpf.sp.gpinf.indexer.search.IPEDSource.openIndex(IPEDSource.java:339)
	at dpf.sp.gpinf.indexer.search.IPEDSource.<init>(IPEDSource.java:197)
	at dpf.sp.gpinf.indexer.search.IPEDSource.<init>(IPEDSource.java:142)
	at dpf.sp.gpinf.indexer.desktop.InicializarBusca.doInBackground(InicializarBusca.java:84)
	at dpf.sp.gpinf.indexer.desktop.InicializarBusca.doInBackground(InicializarBusca.java:41)
	at javax.swing.SwingWorker$1.call(SwingWorker.java:295)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at javax.swing.SwingWorker.run(SwingWorker.java:334)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

Maybe related to conflicting (and repackaged) neo4j lucene version, perhaps eclipse is not following that like maven...

**Comment by lfcnassif** [comment]
I did a testing exclusion of neo4j shaded jar (lucene package was repackaged ok!), then a similar error was thrown from a not repackaged class from the elasticsearch shaded jar (shaded because of lucene version conflicts again), and after removing the last jar, it worked. Strangely the shaded jars don't work only when lauching from eclipse, compiled app works.

@bhoelz do you have some idea about how to solve this?

**Comment by lfcnassif** [comment]
This is the elastic search error:
```
java.lang.NoClassDefFoundError: org/elasticsearch/common/xcontent/ToXContentFragment
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:756)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:468)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:74)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:369)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:363)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:362)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:355)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:264)
	at dpf.sp.gpinf.indexer.process.IndexItem.loadMetadataTypes(IndexItem.java:199)
	at dpf.sp.gpinf.indexer.search.IPEDSource.<init>(IPEDSource.java:226)
	at dpf.sp.gpinf.indexer.search.IPEDSource.<init>(IPEDSource.java:142)
	at dpf.sp.gpinf.indexer.desktop.InicializarBusca.doInBackground(InicializarBusca.java:84)
	at dpf.sp.gpinf.indexer.desktop.InicializarBusca.doInBackground(InicializarBusca.java:41)
	at javax.swing.SwingWorker$1.call(SwingWorker.java:295)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at javax.swing.SwingWorker.run(SwingWorker.java:334)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by lfcnassif** [comment]
First NoSuchMethodError error makes me think eclipse somehow is finding the repackage classes in their original packages, but the not found ToXContentFragment class in the elasticsearch shaded jar is in its original package...

**Comment by lfcnassif** [comment]
This is not related to IPED custom class loader, because it is skipped when debugging from eclipse.

**Comment by lfcnassif** [comment]
My fault, I was with neo4j and elasticsearch original jars on my workspace, used by other projects :(
Commit eac8251 did fix all issues.

**Comment by lfcnassif** [comment]
fixed by eac8251

**Comment by lfcnassif** [comment]
Just a final note: when launching from eclipse, iped custom class loader is not used, so libreoffice jars are not loaded. To test libreoffice viewer, libreoffice jars must be manualy added to classpath in launch configurations.
